v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/ddl/Table.java

```diff
@@ -117,14 +121,19 @@ private void prettyPrintPg(
               .collect(Collectors.joining(", ", "\n\tPRIMARY KEY (", ")")));
     }
     appendable.append("\n)");
-    if (interleaveInParent() != null) {
+    if (interleaveInParent() != null && Objects.equals(interleaveType(), "IN PARENT")) {
       appendable
           .append(" \nINTERLEAVE IN PARENT " + identifierQuote)
           .append(interleaveInParent())
           .append(identifierQuote);
       if (onDeleteCascade()) {
         appendable.append(" ON DELETE CASCADE");
       }
+    } else if (interleaveInParent() != null && Objects.equals(interleaveType(), "IN")) {
+      appendable
+          .append(" \nINTERLEAVE IN " + identifierQuote)
+          .append(interleaveInParent())
```

Comment:
This seems a bit off...this case is for `INTERLEAVE IN` branch right? Why does it contain a reference to `interleaveInParent()`? 

I think you might be reusing stuff, but this is making it confusing. Can we remedy this? For example, can we simplify to have -

`public abstract String interleaveIn();` instead of the `interleaveType` bit? Alternatively, we can keep `interleaveType()` but then we need to rename `interleaveInParent` I think.

Reply:
Renaming this to reduce confusing